### ISSUE TITLE
Stuck while compiling on windows 

### ISSUE THREAD

**Issue by steve-chavez** [mvv/text-printer#1]
OS: Windows x64
GHC: 8.4.4 and 8.6.5

While compiling, the following warning is shown:
```
src\Text\Printer\Integral.hs:656:1: [;1mwarning:[;1m[;1m
Forall'd constraint `Ord �' is not bound in RULE lhs
  Orig bndrs: [�, p, $dOrd_amc3, $dIntegral_amc4, $dPrinter_amc5]
  Orig lhs: let {
              $dPrinter_amcc :: Printer p
              [LclId]
              $dPrinter_amcc = $dPrinter_amc5 } in
            let {
              $dIntegral_amcb :: Integral �
              [LclId]
              $dIntegral_amcb = $dIntegral_amc4 } in
            let {
              $dPositionalSystem_amc9 :: PositionalSystem Binary
              [LclId]
              $dPositionalSystem_amc9
                = Text.Printer.Integral.$fPositionalSystemBinary } in
            let {
              $dReal_aow8 :: Real �
              [LclId]
              $dReal_aow8 = GHC.Real.$p1Integral @ � $dIntegral_amc4 } in
            let {
              $dOrd_aowb :: Ord �
              [LclId]
              $dOrd_aowb = GHC.Real.$p2Real @ � $dReal_aow8 } in
            let {
              $dOrd_amca :: Ord �
              [LclId]
              $dOrd_amca = $dOrd_aowb } in
            number'
              @ Binary
              @ �
              @ p
              $dPositionalSystem_amc9
              $dOrd_amca
              $dIntegral_amcb
              $dPrinter_amcc
  optimised lhs: number'
                   @ Binary
                   @ �
                   @ p
                   $dPositionalSystem_amc9
                   $dOrd_amca
                   $dIntegral_amc4
                   $dPrinter_amc5
```
Then compiling somehow doesn't progress.
Logs can be seen here https://ci.appveyor.com/project/steve-chavez/postgrest/builds/25506383.

text-printer used to work normally on GHC 8.0.2 on windows.

Any pointers would be greatly appreciated.

**Comment by steve-chavez** [comment]
On ubuntu 32 bit I get an error with text-printer as well https://circleci.com/gh/PostgREST/postgrest/3313?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link. Though this one quits.

**Comment by robx** [comment]
The problem also occurs when building text-printer stand-alone with GHC 8.6.5 / stackage lts-13.27: https://ci.appveyor.com/project/robx/text-printer/builds/25781889

Changing the alpha to a doesn't help. Removing the offending SPECIALIZE pragmas does:
https://ci.appveyor.com/project/robx/text-printer/builds/25782098

**Comment by robx** [comment]
same behaviour with GHC 8.4.4 / stackage lts-12.26:

https://ci.appveyor.com/project/robx/text-printer/builds/25782417

**Comment by mvv** [comment]
I have fixed the warning in dd8a404, see if it helps with the compiler hangup.

**Comment by robx** [comment]
Indeed, that fixes it. I'll go ahead and file a bug report with ghc.

**Comment by robx** [comment]
GHC bug report: https://gitlab.haskell.org/ghc/ghc/issues/16917

**Comment by mvv** [comment]
The fix is on Hackage (version 0.5.0.1), I'm closing the issue. BTW, it would help GHC devs if you could provide a reduced test case (start with `number'` and strip all dependencies to other parts of the library).
